lib/virt_autotest/esxi_utils.pm

```diff
@@ -0,0 +1,116 @@
+# without any warranty.
+
+# Summary: Utilities for running ESXi commands
+# Maintainer: Nan Zhang <nan.zhang@suse.com>
+
+package virt_autotest::esxi_utils;
+
+use base Exporter;
+use Exporter;
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+
+our @EXPORT = qw(
+  esxi_vm_get_vmid
+  esxi_vm_power_getstate
+  esxi_vm_power_off
+  esxi_vm_power_on
+  esxi_vm_power_reboot
+  esxi_vm_power_reset
+  esxi_vm_power_shutdown
+  esxi_vm_network_binding
+  esxi_vm_public_ip
+  get_host_timestamp
+  disable_all_clock_synchronization
+);
+
+sub esxi_vm_get_vmid {
+    my $vm_name = shift;
+    my $vim_cmd = "vim-cmd vmsvc/getallvms | grep -i $vm_name | awk -F' ' '{print \$1}'";
+    my (undef, $vmid) = console('svirt')->run_cmd($vim_cmd, domain => 'sshVMwareServer', wantarray => 1);
+    chomp($vmid);
+    return $vmid;
+}
+
+sub esxi_vm_power_getstate {
+    my $vmid    = shift;
+    my $vim_cmd = "vim-cmd vmsvc/power.getstate";
+    my (undef, $power_state) = console('svirt')->run_cmd("$vim_cmd $vmid", domain => 'sshVMwareServer', wantarray => 1);
+    return $power_state;
+}
+
+sub esxi_vm_power_off {
+    my $vmid    = shift;
+    my $vim_cmd = "vim-cmd vmsvc/power.off";
+    return console('svirt')->run_cmd("$vim_cmd $vmid", domain => 'sshVMwareServer', wantarray => 1);
+}
+
+sub esxi_vm_power_on {
+    my $vmid    = shift;
+    my $vim_cmd = "vim-cmd vmsvc/power.on";
+    return console('svirt')->run_cmd("$vim_cmd $vmid", domain => 'sshVMwareServer', wantarray => 1);
+}
+
+sub esxi_vm_power_reboot {
+    my $vmid    = shift;
+    my $vim_cmd = "vim-cmd vmsvc/power.reboot";
+    return console('svirt')->run_cmd("$vim_cmd $vmid", domain => 'sshVMwareServer', wantarray => 1);
+}
+
+sub esxi_vm_power_reset {
+    my $vmid    = shift;
+    my $vim_cmd = "vim-cmd vmsvc/power.reset";
+    return console('svirt')->run_cmd("$vim_cmd $vmid", domain => 'sshVMwareServer', wantarray => 1);
+}
+
+sub esxi_vm_power_shutdown {
+    my $vmid    = shift;
+    my $vim_cmd = "vim-cmd vmsvc/power.shutdown";
+    return console('svirt')->run_cmd("$vim_cmd $vmid", domain => 'sshVMwareServer', wantarray => 1);
+}
+
+sub esxi_vm_network_binding {
+    my $vmid    = shift;
+    my $vim_cmd = "vim-cmd vmsvc/get.environment $vmid | grep vswitch | sed -n 1p | cut -d'\"' -f2";
+    my (undef, $vswitch) = console('svirt')->run_cmd($vim_cmd, domain => 'sshVMwareServer', wantarray => 1);
+    return $vswitch;
+}
+
+sub esxi_vm_public_ip {
+    my $vmid    = shift;
+    my $vim_cmd = "vim-cmd vmsvc/get.guest $vmid | grep ipAddress | sed -n 1p | cut -d'\"' -f2";
+    my (undef, $vm_ip) = console('svirt')->run_cmd($vim_cmd, domain => 'sshVMwareServer', wantarray => 1);
+    return $vm_ip;
+}
+
+sub get_host_timestamp {
+    my (undef, $host_time) = console('svirt')->run_cmd("date -u +'\%Y-\%m-\%d \%H:\%M:\%S'", domain => 'sshVMwareServer', wantarray => 1);
+    return $host_time;
+}
+
+sub disable_all_clock_synchronization {
```

Comment:
Do I get it correctly that this function disable clock sync on a single vm? If yes, suggest to spin vm into the function name.

Reply:
Updated the function name with "disable_vm_time_synchronization".